src/model.js

```diff
@@ -0,0 +1,37 @@
+/** @babel */
+
+let nextInstanceId = 1
+
+export default class Model {
+  static resetNextInstanceId () {
+    nextInstanceId = 1
+  }
+
+  constructor (params = {}) {
+    this.alive = true
+    this.assignId(params.id)
+  }
+
+  assignId (id) {
+    this.id = this.id || id || nextInstanceId++
```

Comment:
I don't think it will since we initialize it to `1`, but that's a good point that this could be more robust